### PR TITLE
Rework dash-renderer setProps

### PR DESCRIPTION
**Pull request by Marc-Andre-Rivet**

### CODE REVIEW

File: dash_core_components/package.json [hunk 1]
```diff
@@ -73,6 +73,7 @@
     "prettier": "^1.14.2",
     "react": "^16.6.1",
     "react-dom": "^16.6.1",
+    "react-dot-fragment": "^0.2.8",
```
Comment (Marc-Andre-Rivet) [inline dash_core_components/package.json]: React 16 ```<Fragment />``` shim

File: src/components/ConfirmDialogProvider.react.js [hunk 2]
```diff
@@ -31,20 +32,18 @@ export default class ConfirmDialogProvider extends React.Component {
...
-            });
+        const wrapClick = child => {
+            const props = clone(child.props);
+            props.__dashlayout__.props.onClick = () => {
+                const update = {displayed: true};
+                this.setState(update);
+                if (setProps) {
+                    setProps(update);
+                }
+            };
```
Comment (Marc-Andre-Rivet) [inline src/components/ConfirmDialogProvider.react.js]: Like before, inserting `onClick` prop for the real child to use

File: src/components/ConfirmDialogProvider.react.js [hunk 3]
```diff
@@ -53,9 +52,9 @@ export default class ConfirmDialogProvider extends React.Component {
                     (loading_state && loading_state.is_loading) || undefined
                 }
             >
-                {realChild && realChild.length
-                    ? realChild.map(wrapClick)
-                    : wrapClick(realChild)}
+                {children && children.length
+                    ? children.map(c => wrapClick(c))
+                    : wrapClick(children)}
```
Comment (Marc-Andre-Rivet) [inline src/components/ConfirmDialogProvider.react.js]: The child is not nested -- operate on the tree container directly

Comment (alexcjohnson) [reply src/components/ConfirmDialogProvider.react.js]: Why the change from `children.map(wrapClick)` to `children.map(c => wrapClick(c))`?

Also, not new but perhaps `Array.isArray(children)` would be better than `children && children.length`.

File: test/unit/Tabs.test.js [hunk 4]
```diff
@@ -177,12 +177,12 @@ describe('Tabs handle Tab selection logic', () => {
...
     test('Tab can be clicked and will display its content', () => {
         tabs.find('[value="tab-2"]').simulate('click');
-        const renderedContent = tabs.find('.tab-content > div').html();
+        const renderedContent = tabs.find('.tab-content div').html();
         expect(renderedContent).toEqual('<div>Tab 2 child</div>');
     });
     test('Tab without value will still be clickable', () => {
         tabs.find('[value="tab-2"]').simulate('click');
-        const renderedContent = tabs.find('.tab-content > div').html();
+        const renderedContent = tabs.find('.tab-content div').html();
```
Comment (alexcjohnson) [inline test/unit/Tabs.test.js]: Is this change still necessary? Or did `<Fragment>` restore the previous structure?

File: src/components/ConfirmDialogProvider.react.js (hunk 3 above)
Comment (Marc-Andre-Rivet) [reply src/components/ConfirmDialogProvider.react.js]: Guessing I did a console.log at some point and didn't revert it back correctly. Reverting.

File: test/unit/Tabs.test.js [hunk 5]
```diff
@@ -177,12 +177,12 @@ describe('Tabs handle Tab selection logic', () => {
     });
     test('Tab can be clicked and will display its content', () => {
         tabs.find('[value="tab-2"]').simulate('click');
-        const renderedContent = tabs.find('.tab-content > div').html();
+        const renderedContent = tabs.find(Tab).html();
```
Comment (Marc-Andre-Rivet) [inline test/unit/Tabs.test.js]: Checking for the Tab component instead of some arbitrary selector

File: test/unit/Tabs.test.js [hunk 6]
```diff
@@ -177,12 +177,12 @@ describe('Tabs handle Tab selection logic', () => {
...
         expect(renderedContent).toEqual('<div>Tab 2 child</div>');
     });
     test('Tab without value will still be clickable', () => {
         tabs.find('[value="tab-2"]').simulate('click');
-        const renderedContent = tabs.find('.tab-content > div').html();
+        const renderedContent = tabs.find(Tab).html();
```
Comment (Marc-Andre-Rivet) [inline test/unit/Tabs.test.js]: This test makes no sense. It's the exact copy of the one above with a different name. It's clearly not testing that case.. removing.

File: test/unit/Tabs.test.js [hunk 7]
```diff
@@ -177,12 +177,7 @@ describe('Tabs handle Tab selection logic', () => {
     });
     test('Tab can be clicked and will display its content', () => {
         tabs.find('[value="tab-2"]').simulate('click');
-        const renderedContent = tabs.find('.tab-content > div').html();
-        expect(renderedContent).toEqual('<div>Tab 2 child</div>');
-    });
-    test('Tab without value will still be clickable', () => {
-        tabs.find('[value="tab-2"]').simulate('click');
-        const renderedContent = tabs.find('.tab-content > div').html();
```
Comment (Marc-Andre-Rivet) [inline test/unit/Tabs.test.js]: This test is an exact copy of the test above. Looks like copy-paste that was never modified. Removing.

File: test/unit/Tabs.test.js (hunk 4 above)
Comment (Marc-Andre-Rivet) [reply test/unit/Tabs.test.js]: See final change below
